storm-core/src/jvm/org/apache/storm/daemon/DrpcServer.java

```diff
@@ -0,0 +1,390 @@
+            if (sem != null) {
+                this.idtoResult.put(id, result);
+                sem.release();
+            }
+        }
+    }
+
+    @Override
+    public DRPCRequest fetchRequest(String functionName) throws AuthorizationException, TException {
+        meterFetchRequestCalls.mark();
+        Map<String, String> map = new HashMap<>();
+        map.put(DRPCAuthorizerBase.FUNCTION_NAME, functionName);
+        checkAuthorization(authorizer, map, "fetchRequest");
+        ConcurrentLinkedQueue<DRPCRequest> queue = acquireQueue(functionName);
+        DRPCRequest req = queue.poll();
+        if (req != null) {
+            LOG.debug("Fetched request for {} at {}", functionName, System.currentTimeMillis());
+            return req;
+        } else {
+            return new DRPCRequest("", "");
+        }
+    }
+
+    @Override
+    public void failRequest(String id) throws AuthorizationException, TException {
+        meterFailRequestCalls.mark();
+        String func = this.idtoFunction.get(id);
+        if (func != null) {
+            Map<String, String> map = new HashMap<>();
+            map.put(DRPCAuthorizerBase.FUNCTION_NAME, func);
+            checkAuthorization(authorizer, map, "failRequest");
+            Semaphore sem = this.idtoSem.get(id);
+            if (sem != null) {
+                this.idtoResult.put(id, new DRPCExecutionException("Request failed"));
+                sem.release();
+            }
+        }
+    }
+
+    protected ConcurrentLinkedQueue<DRPCRequest> acquireQueue(String function) {
+        ConcurrentLinkedQueue<DRPCRequest> reqQueue = requestQueues.get(function);
+        if (reqQueue == null) {
+            reqQueue = new ConcurrentLinkedQueue<DRPCRequest>();
+            requestQueues.put(function, reqQueue);
+        }
+        return reqQueue;
+    }
+
+    private void checkAuthorization(IAuthorizer aclHandler, Map mapping, String operation, ReqContext reqContext) throws AuthorizationException {
+        if (reqContext != null) {
+            ThriftAccessLogger.logAccess(reqContext.requestID(), reqContext.remoteAddress(), reqContext.principal(), operation);
+        }
+        if (aclHandler != null) {
+            if (reqContext == null)
+                reqContext = ReqContext.context();
+            if (!aclHandler.permit(reqContext, operation, mapping)) {
+                Principal principal = reqContext.principal();
+                String user = (principal != null) ? principal.getName() : "unknown";
+                throw new AuthorizationException("DRPC request '" + operation + "' for '" + user + "' user is not authorized");
+            }
+        }
+    }
+
+    private void checkAuthorization(IAuthorizer aclHandler, Map mapping, String operation) throws AuthorizationException {
+        checkAuthorization(aclHandler, mapping, operation, ReqContext.context());
+    }
+
+    // TO be replaced by Common.mkAuthorizationHandler
```

Comment:
Then lets create the file and just put it in there now, so the next person does not have to do it later.